azure-mgmt-cdn/src/main/java/com/microsoft/azure/management/cdn/CdnEndpoint.java

```diff
@@ -0,0 +1,445 @@
+package com.microsoft.azure.management.cdn;
+
+import com.microsoft.azure.management.apigeneration.Fluent;
+import com.microsoft.azure.management.cdn.implementation.EndpointInner;
+import com.microsoft.azure.management.resources.fluentcore.arm.CountryISOCode;
+import com.microsoft.azure.management.resources.fluentcore.arm.models.ExternalChildResource;
+import com.microsoft.azure.management.resources.fluentcore.model.Settable;
+import com.microsoft.azure.management.resources.fluentcore.model.Wrapper;
+
+import java.util.List;
+
+/**
+ * An immutable client-side representation of an Azure CDN endpoint.
+ */
+@Fluent
+public interface CdnEndpoint extends
+        ExternalChildResource<CdnEndpoint, CdnProfile>,
+        Wrapper<EndpointInner> {
+
+    /**
+     * Get the originHostHeader value.
+     *
+     * @return the originHostHeader value
+     */
+    String originHostHeader();
+
+    /**
+     * Get the originPath value.
+     *
+     * @return the originPath value
+     */
+    String originPath();
+
+    /**
+     * Get the contentTypesToCompress value.
+     *
+     * @return the contentTypesToCompress value
+     */
+    List<String> contentTypesToCompress();
+
+    /**
+     * Get the isCompressionEnabled value.
+     *
+     * @return the isCompressionEnabled value
+     */
+    boolean isCompressionEnabled();
+
+    /**
+     * Get the isHttpAllowed value.
+     *
+     * @return the isHttpAllowed value
+     */
+    boolean isHttpAllowed();
+
+    /**
+     * Get the isHttpsAllowed value.
+     *
+     * @return the isHttpsAllowed value
```

Comment:
for the Boolean, one descriptive way would be to say "true is HTTP is allowed, else false"